src/qtism/data/storage/xml/filesystem/FlysystemV2Filesystem.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace qtism\data\storage\xml\filesystem;
+
+use Exception;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FilesystemException as FlysystemException;
+use League\Flysystem\Local\LocalFilesystemAdapter;
+
+class FlysystemV2Filesystem implements FilesystemInterface
+{
+    protected Filesystem $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    public static function local(string $path = '/'): self
+    {
+        return new self(new Filesystem(new LocalFilesystemAdapter($path)));
```

Comment:
Bear in mind that `XmlDocument` is a serializable object, and to keep it that way, `LocalFilesystemAdapter` instance (being composed into that object) will need to be serializable as well.

By default, though, `LocalFilesystemAdapter` will instantiate a `FinfoMimeTypeDetector` [here](https://github.com/thephpleague/flysystem/blob/3.x/src/Local/LocalFilesystemAdapter.php#L80), which will make the whole composition non-serializable due to [the `finfo` object](https://github.com/thephpleague/mime-type-detection/blob/main/src/FinfoMimeTypeDetector.php#L48).

Reply:
It isn't clear that `XmlDocument` is intended to be `Serializable`. What's the use case of XmlDocument being serialized?  Can the filesystem be excluded during serialization and require it be re-set when unserialized? The current implementation suffers from the same potential issue as there's no implied expectation that `Filesystem` must also be serializable, and it seems a bad idea anyway to include the associated filesystem implementation in the serialized content.

Reply:
You're right on so many levels. However, it still can be kept serializable if https://github.com/thephpleague/mime-type-detection/blob/main/src/ExtensionMimeTypeDetector.php gets injected into the LocalFilesystemAdapter instance here by default.

In fact, that one should probably be sufficient for most of the use cases, as I believe the assets within a QTI package usually have their extensions in place.